1_stock_prices/stock_prices.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/python
+
+import math
+import argparse
+
+# References
+# https://medium.com/@antash/six-ways-to-find-max-value-of-a-list-in-python-b7d7ccfabc0d
+
+
+# How can I turn a list into an object?
+# How can I iterate over an object?
+
+# def find_max_profit(prices):
+#   max = 0
+#   min = math.inf
+#   prices = [int(x) for x in prices]
+#   for price in prices:
+#     if price > max:
+#       max = price
+#   for price in prices:
+#     if price < max:
+#       min = price
+#   return max-min
+
+
+def find_max_profit(prices):
+  return max(prices) - min(prices)
+
+
+print(find_max_profit([10, 7, 5, 8, 11, 9]))
+
```

Comment:
Does your solution work in all test cases? Try running `test_stock_prices.py` and see the results.

This is what I wrote when I was explaining the requirements of this to another student:
> Do you get why `find_max_profit([1050, 270, 1540, 3800, 2])` is `3530`?
> So let's pretend those 5 numbers are stock prices from the last 5 months:
Jan: 1050
Feb: 270
Mar: 1540
Apr: 3800
May: 2
>
> The lowest price is `2`, but you can't time travel back to Apr (or engage in some exotic financial sorcery) to sell it for `3800`. That's why `3798` is _not_ the answer
> Instead, buying at Feb for `270` and selling at Apr for `3800` yields the _highest profit_

I believe your current implementation will return the correct answer, `11`, for your current test case of `[10, 7, 5, 8, 11, 9]`, but will incorrectly give `3798`, instead of `3530`, given a list of `[1050, 270, 1540, 3800, 2]`.

Reply:
You're right.  I need to revisit this and modify the implementation!  I wasn't understanding the timing aspect of the stock prices.